File: cpp/difference-of-squares/difference_of_squares.cpp
#include "difference_of_squares.h"

int difference_of_squares::square_of_sum(int num)
{
	int sum = 0;

	for (int i = 1; i <= num; i++)
		sum += i;

	return sum * sum;
}

int difference_of_squares::sum_of_squares(int num)
{
	int sum = 0;

	for (int i = 1; i <= num; i++)
		sum += i * i;
	
	return sum;
}

int difference_of_squares::difference(int num) 
{
	return square_of_sum(num) - sum_of_squares(num);
}

File: go/difference-of-squares/difference_of_squares.go
// Package diffsquares contains mathematical methods to square sums, sum squares, etc.
package diffsquares

// SquareOfSum calculates the square of the sum from [1, num].
func SquareOfSum(num int) int {
	sum := (num * (num + 1)) / 2

	return sum * sum
}

// SumOfSquares calculates the sum of all the squares from [1, num].
func SumOfSquares(num int) int {
	return (num * (num + 1) * (2*num + 1)) / 6
}

// Difference finds the difference between SquareOfSum(num) and SumOfSquares(num).
func Difference(num int) int {
	return SquareOfSum(num) - SumOfSquares(num)
}
